code/modules/transport/tram/tram_structures.dm

```diff
@@ -550,6 +580,43 @@
 
 	return FALSE
 
+/obj/structure/tram/spoiler/welder_act(mob/living/user, obj/item/tool)
+	if(!tool.tool_start_check(user, amount = 1))
+		return FALSE
+
+	if(atom_integrity >= max_integrity)
+		if(malfunctioning)
+			to_chat(user, span_warning("You begin to repair \the [src], allowing movement."))
+			if(tool.use_tool(src, user, 4 SECONDS, volume = 50))
+				malfunctioning = FALSE
+				user.visible_message(span_warning("[user] repairs the damage to \the [src] with [tool]."), \
+					span_warning("You unweld \the [src], it can move freely again!"), null, COMBAT_MESSAGE_RANGE)
+				update_appearance()
+
+		else
+			to_chat(user, span_warning("You begin to weld \the [src], preventing retraction."))
+			if(tool.use_tool(src, user, 4 SECONDS, volume = 50))
+				malfunctioning = TRUE
+				user.visible_message(span_warning("[user] locks \the [src] in place with [tool]."), \
+					span_warning("You weld \the [src], it will remain deployed!"), null, COMBAT_MESSAGE_RANGE)
+				deploy_spoiler()
+				update_appearance()
+
+		return TOOL_ACT_TOOLTYPE_SUCCESS
```

Comment:
Do we want to return this if we did not succeed?
Might want to invert the `use_tool` checks to early return on failure and _not_ return a success there?

Reply:
Yeah that works better, can remove some duplication as well.